js/summary.js

```diff
@@ -116,12 +128,18 @@ H5P.Summary = (function ($, Question, JoubelUI) {
 
     // Register retry button to h5p question
     var self = this;
-    this.addButton('try-again', 'retry', function () {
 
-      // Remove button after is has been clicked
-      self.hideButton('try-again');
-      self.resetTask();
-    }, false);
+    // Attach the retry buton if it is enabled in the semantics
+    if (this.options.behaviour.enableRetry) {
+
+      // Add JoubelUI retry button using the h5p-question module
+      this.addButton('try-again', 'retry', function () {
+
+        // Remove button after is has been clicked
+        self.hideButton('try-again');
+        self.resetTask();
+      }, self.progress == self.summaries.length);
```

Comment:
Another note regarding self.summaries.length:
When adding the summary as a sub-contet-type, for instance inside Course Presentation, an "empty shell" is initialized without any summaries parameters. Your code expect self.summaries.length to be defined when adding the 'try-again' button, but this may not be the case. You need to check whether it is defined first.